Trifolia.Web/Scripts/TemplateEdit/TemplateEditViewModel.js

```diff
@@ -24,6 +24,113 @@ var templateEditViewModel = function (templateId, defaults) {
     self.AvailableExtensions = ko.observableArray([]);
     self.SelectedAvailableExtensionId = ko.observable();
 
+    self.duplicateNode = function () {
+        if (!self.CurrentNode() || !self.CurrentNode().Constraint()) {
+            return false;
+        }
+
+        var constraint = self.CurrentNode().Constraint();
+
+        if (!constraint.IsPrimitive()) {
+            for (var x = 0; x < self.Constraints().length; x++) {
+                if (constraint.Context() === self.Constraints()[x].Context() && constraint.Id() !== self.Constraints()[x].Id()) {
+                    return true;
+                }
+            }
+        } else {
+            var siblings = constraint.Parent().Children();
+            for (var x = 0; x < siblings.length; x++) {
```

Comment:
Not required... but, you could consider using _ (underscore.js) here. I have recently added it to the library of scripts used by Trifolia. I could see _.first() being used here to determine if it should return true.